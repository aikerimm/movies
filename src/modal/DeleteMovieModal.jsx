import PropTypes from 'prop-types';
import './deleteMovieModal.css';
import Button from '../util/Button';

const DeleteMovieModal = ({ open, onClose, movieId, onDelete }) => {
  return (
    open && (
      <div className='overlay' onClick={onClose}>
        <div
          className='deleteMovieModal'
          onClick={(event) => event.stopPropagation()}
        >
          <p className='closeBtn' onClick={onClose}>
            X
          </p>
          <p className='titleText fullWidth'>Delete movie</p>
          <p className='promptText'>
            Are you sure you want to delete this movie
          </p>
          <div className='deleteConfirmDiv'>
            <Button
              value='confirm'
              type='submit'
              onClick={() => onDelete(movieId)}
            />
          </div>
        </div>
      </div>
    )
  );
};

DeleteMovieModal.defaultProps = {
  open: false,
};

DeleteMovieModal.propTypes = {
  open: PropTypes.bool,
  onClose: PropTypes.func.isRequired,
  movieId: PropTypes.number.isRequired,
  onDelete: PropTypes.func.isRequired,
};

export default DeleteMovieModal;
