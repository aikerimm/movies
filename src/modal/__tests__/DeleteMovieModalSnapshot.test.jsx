import renderer from 'react-test-renderer';
import DeleteMovieModal from '../DeleteMovieModal.jsx';

it('DeleteMovieModal should render correctly', () => {
  const tree = renderer.create(<DeleteMovieModal open={true} onClose={jest.fn()} onDelete={jest.fn()} movieId={123}/>).toJSON();
  expect(tree).toMatchSnapshot();
});