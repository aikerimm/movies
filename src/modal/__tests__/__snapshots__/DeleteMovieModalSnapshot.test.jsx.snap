// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeleteMovieModal should render correctly 1`] = `
<div
  className="overlay"
  onClick={[MockFunction]}
>
  <div
    className="deleteMovieModal"
    onClick={[Function]}
  >
    <p
      className="closeBtn"
      onClick={[MockFunction]}
    >
      X
    </p>
    <p
      className="titleText fullWidth"
    >
      Delete movie
    </p>
    <p
      className="promptText"
    >
      Are you sure you want to delete this movie
    </p>
    <div
      className="deleteConfirmDiv"
    >
      <input
        className="submitBtn roundedCorners"
        onClick={[Function]}
        type="button"
        value="confirm"
      />
    </div>
  </div>
</div>
`;
